pkg/web_app/lib/src/account.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:html';
+import 'dart:js';
+
+import 'package:http/browser_client.dart';
+import 'package:http/http.dart';
+
+import 'google_auth_js.dart';
+
+Element _logoutElem;
+Element _accountInfoElem;
+
+GoogleUser _currentUser;
+BrowserClient client = BrowserClient();
+
+void _updateUser(GoogleUser user) {
+  final isSignedIn = getAuthInstance().isSignedIn.get();
+  _currentUser = !isSignedIn || user?.getId() == null ? null : user;
+  client.close();
+  if (_currentUser == null) {
+    print('No active user');
+    client = BrowserClient();
+    _accountInfoElem.text = '-';
+  } else {
+    print('user: ${_currentUser.getBasicProfile().getEmail()}');
+    client =
+        AuthenticatedClient(_currentUser.getAuthResponse(true)?.access_token);
+    _updateOnCredChange();
+  }
+  _updateVisibility();
+  print('updated user.');
+}
+
+Future<void> setupAccount() async {
+  context['pubAuthInit'] = () {
+    if (getAuthInstance().isSignedIn.get()) {
```

Comment:
Don't you still have to do:
```
  gapi.load('auth2', function() { // Ready. });
```
See: https://developers.google.com/identity/sign-in/web/reference

Reply:
This is a callback that will be executed once the script loads and auth is initialized. (it is triggered via the request param in the script)

Reply:
This is equivalent of the `init` function specified in:
```
<script src="https://apis.google.com/js/platform.js?onload=init" async defer></script>
```
From the docs... where do you get that you don't need to call `gapi.load('auth2',` ?

Reply:
This is exactly what we are doing, I just named it `pubAuthInit` instead of `init`.

`gapi.load(...)`: not sure, but I think there was a half-sentence about the `onload` callback will be called after the gapi is loaded/initialized.

Reply:
It's very possible that `gapi` is loaded, but we also have to load the `auth2` library.

I suspect it's working because our browsers have all this stuff cached...
Regardless of why it works without this step, we should include it as to follow documentation so that it keeps working in the future :)

Reply:
done (in the other PR)